### PR TITLE
[Mac] Use INSWindowDelegate instead of NSWindow events

### PR DESCRIPTION
**Pull request by sevoku**
Fixes VSTS #935131

### CODE REVIEW

File: Xwt.XamMac/Xwt.Mac/WindowBackend.cs [hunk 1]
```diff
@@ -211,33 +210,45 @@ void IBackend.EnableEvent (object eventId)
 			if (eventId is WindowFrameEvent) {
 				var @event = (WindowFrameEvent)eventId;
 				switch (@event) {
-				case WindowFrameEvent.BoundsChanged:
```
Comment (Therzok) [inline Xwt.XamMac/Xwt.Mac/WindowBackend.cs]: Shown/Hidden uses observers and this uses selectors. Should we unify on one pattern? There's DidResizeNotification and DidMoveNotification.

File: Xwt.XamMac/Xwt.Mac/WindowBackend.cs [hunk 2]
```diff
@@ -211,33 +210,45 @@ void IBackend.EnableEvent (object eventId)
...
-					EnableVisibilityEvent (@event);
-					break;
-				case WindowFrameEvent.CloseRequested:
-					WindowShouldClose = OnShouldClose;
-					break;
+					case WindowFrameEvent.Hidden:
+					case WindowFrameEvent.Shown:
+						if (!VisibilityEventsEnabled())
+						{
+							ContentView.AddObserver(this, HiddenProperty, NSKeyValueObservingOptions.New, IntPtr.Zero);
```
Comment (Therzok) [inline Xwt.XamMac/Xwt.Mac/WindowBackend.cs]: Store the observer and dispose it on window dispose.

File: Xwt.XamMac/Xwt.Mac/WindowBackend.cs (hunk 1 above)
Comment (sevoku) [reply Xwt.XamMac/Xwt.Mac/WindowBackend.cs]: Using selectors with the delegate is cleaner imo, so I'd prefer keeping it. Unfortunately we can't change thjat for the observer cos we observe ContentView and not the window.

File: Xwt.XamMac/Xwt.Mac/WindowBackend.cs (hunk 2 above)
Comment (sevoku) [reply Xwt.XamMac/Xwt.Mac/WindowBackend.cs]: hhm, `AddObserver` is `void` (it's not like NotificationManager) and we remove it here: https://github.com/mono/xwt/blob/512728f98e8390de4919b85ee7177a8e1b35994b/Xwt.XamMac/Xwt.Mac/WindowBackend.cs#L477

Comment (Therzok) [reply Xwt.XamMac/Xwt.Mac/WindowBackend.cs]: Ah, I see. Good.

**Pull request merged**